### PR TITLE
[SPARK-9528] [ml] Changed RandomForestClassifier to extend ProbabilisticClassifier

### PR DESCRIPTION
**Pull request by jkbradley**
RandomForestClassifier now outputs rawPrediction based on tree probabilities, plus probability column computed from normalized rawPrediction.

CC: @holdenk

### CODE REVIEW

**Comment by jkbradley** [comment]
@yanboliang If you have a chance, would you mind taking a look at this?  Thanks!

File: mllib/src/main/scala/org/apache/spark/ml/classification/RandomForestClassifier.scala [hunk 1]
```diff
@@ -17,22 +17,19 @@
...
-import org.apache.spark.mllib.linalg.{Vector, Vectors}
+import org.apache.spark.mllib.linalg.{SparseVector, DenseVector, Vector, Vectors}
 import org.apache.spark.mllib.regression.LabeledPoint
 import org.apache.spark.mllib.tree.configuration.{Algo => OldAlgo}
 import org.apache.spark.mllib.tree.model.{RandomForestModel => OldRandomForestModel}
 import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.DataFrame
 import org.apache.spark.sql.functions._
-import org.apache.spark.sql.types.DoubleType
+
```
Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/ml/classification/RandomForestClassifier.scala]: Redundant blank line

**Comment by yanboliang** [comment]
The same method as sklearn RF [predict_proba](https://github.com/scikit-learn/scikit-learn/blob/0e87eb8/sklearn/ensemble/forest.py#L510), so looks well.

**Comment by jkbradley** [comment]
@yanboliang Thank you for taking a look!  I'll merge this with master.
